runtime/compilers/rillv1/parse_rillyaml.go

```diff
@@ -178,3 +181,76 @@ func (p *Parser) parseRillYAML(ctx context.Context, path string) error {
 	p.RillYAML = res
 	return nil
 }
+
+func equalRillYAML(r, other *RillYAML) bool {
```

Comment:
1. Consider adding as a member function: `(r *RillYAML) equals(other *RillYAML) bool`
2. Did you check if using `reflect.DeepEquals` would do the job? We don't need a very good comparison function here, just avoiding reparses for the CHMOD (where the YAML is exactly the same) is fine. We could also consider just storing an MD5 hash of the YAML contents in the struct and comparing that.

Reply:
No I did not evaluate `reflect.DeepEquals` earlier. 
Checked and seems to work fine.